src/components/street-segment.js

```diff
@@ -0,0 +1,424 @@
+    <a-entity street-segment="type: sidewalk; surface: concrete; color: white; width: 6; length: 150"></a-entity>
+</a-entity>
+*/
+
+const COLORS = {
+  red: '#ff9393',
+  blue: '#00b6b6',
+  green: '#adff83',
+  yellow: '#f7d117',
+  lightGray: '#dddddd',
+  white: '#ffffff',
+  brown: '#664B00'
+};
+STREET.colors = COLORS;
+
+const TYPES = {
+  'drive-lane': {
+    type: 'drive-lane',
+    color: COLORS.white,
+    surface: 'asphalt',
+    level: 0,
+    generated: {
+      clones: [
+        {
+          mode: 'random',
+          modelsArray:
+            'sedan-rig, box-truck-rig, self-driving-waymo-car, suv-rig, motorbike',
+          spacing: '7.3',
+          count: '4'
+        }
+      ]
+    }
+  },
+  'bus-lane': {
+    type: 'bus-lane',
+    surface: 'asphalt',
+    color: COLORS.red,
+    level: 0,
+    generated: {
+      clones: [
+        {
+          mode: 'random',
+          model: 'bus',
+          spacing: '15',
+          count: '1'
+        }
+      ],
+      stencil: [
+        {
+          stencils: 'word-only, word-taxi, word-bus',
+          spacing: '40',
+          padding: '10'
+        }
+      ]
+    }
+  },
+  'bike-lane': {
+    type: 'bike-lane',
+    color: COLORS.green,
+    surface: 'asphalt',
+    level: 0,
+    generated: {
+      stencil: [
+        {
+          model: 'bike-arrow',
+          cycleOffset: '0.3',
+          spacing: '20'
+        }
+      ],
+      clones: [
+        {
+          mode: 'random',
+          modelsArray:
+            'cyclist-cargo, cyclist1, cyclist2, cyclist3, cyclist-dutch, cyclist-kid, ElectricScooter_1',
+          spacing: '2.03',
+          count: '4'
+        }
+      ]
+    }
+  },
+  sidewalk: {
+    type: 'sidewalk',
+    surface: 'sidewalk',
+    color: COLORS.white,
+    level: 1,
+    direction: 'none',
+    generated: {
+      pedestrians: [
+        {
+          density: 'normal'
+        }
+      ]
+    }
+  },
+  'parking-lane': {
+    surface: 'concrete',
+    color: COLORS.lightGray,
+    level: 0,
+    generated: {
+      clones: [
+        {
+          mode: 'random',
+          modelsArray: 'sedan-rig, self-driving-waymo-car, suv-rig',
+          spacing: 6,
+          count: 6
+        }
+      ],
+      stencil: [
+        {
+          model: 'parking-t',
+          cycleOffset: 1,
+          spacing: 6
+        }
+      ]
+    }
+  },
+  divider: {
+    surface: 'hatched',
+    color: COLORS.white,
+    level: 0
+  },
+  grass: {
+    surface: 'grass',
+    color: COLORS.white,
+    level: -1
+  },
+  rail: {
+    surface: 'asphalt',
+    color: COLORS.white,
+    level: 0
+  }
+};
+STREET.types = TYPES;
+
+AFRAME.registerComponent('street-segment', {
+  schema: {
+    type: {
+      type: 'string', // value not used by component, used in React app instead
+      oneOf: [
+        'drive-lane',
+        'bus-lane',
+        'bike-lane',
+        'sidewalk',
+        'parking-lane',
+        'divider',
+        'grass',
+        'rail'
+      ]
```

Comment:
c9b4908afb68e31019b9b94a8e56faa47d0f1b79

Reply:
reverted this, for some reason the dropdown doesn't work correctly when using `oneOf: Object.keys(TYPES)` instead of listing them out manually

![image](https://github.com/user-attachments/assets/febb4bfe-28e7-4fae-a0f2-999bfd324fc4)

Reply:
That's because you really tested `oneOf: [Object.keys(TYPES)]` and not `oneOf: Object.keys(TYPES)` as you said.